Adding Facebook sample app (#6220)
AzureAD/microsoft-authentication-library-for-js: samples/msal-browser-samples/VanillaJSTestApp2.0/app/facebook-sample/Readme.md

# MSAL.js Sample - Authorization Code Flow for Non-Microsoft Identity Providers

## About this sample
This developer sample is used to demonstrate how to configure an app for use with non-Microsoft identity providers by using Facebook as an example.

## How to run the sample:
- Replace client ID with the app ID from the basic settings page of the app registration on the [Meta for Developers page](https://developers.facebook.com/).

## Important changes when using non-Microsoft identity providers
- You must change the authority in the config to an authority supported by the identity provider. Additionally, for non-Microsoft authorities, you must add the authority (without the https://) to the knownAuthorities parameter in the config.
- You must set the protocol mode to OIDC.
- You have the option of configuring different authentication options when using OIDC protocol mode. These are set in the OIDCOptions parameter. 

>     - serverResponseType sets the form in which MSAL requests and expects server responses. The options are ServerResponseType.FRAGMENT (for a hash fragment) or ServerResponseType.QUERY (for a query parameter). If not set, it defaults to ServerResponseType.FRAGMENT. If supported by the identity provider, we highly recommend using the ServerResponseType.FRAGMENT.

Reviewer: ```suggestion
    - serverResponseType sets the format supported by MSAL for requests and server responses. Supported options are:  `ServerResponseType.FRAGMENT` (for a hash fragment) or ServerResponseType.QUERY (for a query parameter). If not set, it defaults to ServerResponseType.FRAGMENT. If supported by the identity provider, we highly recommend using the ServerResponseType.FRAGMENT.
```

>     - serverResponseType sets the form in which MSAL requests and expects server responses. The options are ServerResponseType.FRAGMENT (for a hash fragment) or ServerResponseType.QUERY (for a query parameter). If not set, it defaults to ServerResponseType.FRAGMENT. If supported by the identity provider, we highly recommend using the ServerResponseType.FRAGMENT.

Reviewer 2: Can we add a table with the different `serverResponseType` values? I think it'd be clearer.

Author: I couldn't think of anything to add for a third column (which is required for a table), but I moved it to bullet points so that it might be clearer.

>     - defaultScopes provides the option to override the default scopes sent by MSAL. If not set, it defaults to ["openid", "profile", "offline_access"]. If defaultScopes does not include "openid", MSAL will automatically add it for OIDC compliance.

Reviewer: ```suggestion
    - The `defaultScopes` parameter provides the option to override the default scopes sent by MSAL. If not set, it defaults to ["openid", "profile", "offline_access"]. If `defaultScopes` does not include "openid", MSAL will automatically add it for OIDC compliance.
```

> - You have the option of manually configuring endpoints used by the identity provider. If not set, MSAL will attempt to discover the endpoints. Manual endpoint configuration should be passed in a stringified JSON object and include issuer, authorization_endpoint, token_endpoint, jwks_uri, and (if available) end_session_endpoint.

Reviewer: ```suggestion
- You have the option to manually configure endpoints used by the identity provider. If not set, MSAL will attempt to discover the endpoints. Manual endpoint configuration should be passed in a stringified JSON object and include issuer, authorization_endpoint, token_endpoint, jwks_uri, and (if available) end_session_endpoint.
```

    - The `serverResponseType` parameter sets the format supported by MSAL for requests and server responses. Supported options are:
        - `ServerResponseType.FRAGMENT` (for a hash fragment). If not set, MSAL defaults to `ServerResponseType.FRAGMENT`. If supported by the identity provider, we highly recommend using `ServerResponseType.FRAGMENT`.
        - `ServerResponseType.QUERY` (for a query parameter). We recommend using `ServerResponseType.QUERY` only if the identity provider does not support sending server responses in a hash fragment.
    - The `defaultScopes` parameter provides the option to override the default scopes sent by MSAL. If not set, it defaults to ["openid", "profile", "offline_access"]. If `defaultScopes` does not include "openid", MSAL will automatically add it for OIDC compliance.
- You have the option to manually configure endpoints used by the identity provider. If not set, MSAL will attempt to discover the endpoints. Manual endpoint configuration should be passed in a stringified JSON object and include issuer, authorization_endpoint, token_endpoint, jwks_uri, and (if available) end_session_endpoint.

> - You must set allowNativeBroker to false when using a non-Microsoft identity provider.

Reviewer 3: I think this is more in-line with the desired behavior. Not providing any value and providing false should both be acceptable. If protocolMode is not AAD, we should default this to false, if protocolMode is not AAD and someone explicitly passes true we should throw (I believe your other PR implements this)

```suggestion
- You must not set allowNativeBroker to true when using a non-Microsoft identity provider.
```
